src/Microsoft.VisualBasic/tests/VB/DriveInfo.Windows.Tests.vb

```diff
@@ -0,0 +1,174 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.VisualBasic.FileIO
+
+Imports System
+Imports System.Collections
+Imports System.Collections.Generic
+Imports System.Linq
+Imports System.Runtime.InteropServices
+Imports System.Security
+Imports Xunit
+Imports System.Text
+' DO NOT IMPORT System.IO
+Namespace Microsoft.VisualBasic.Tests
+
+    Public NotInheritable Class DriveInfoWindowsTests
+        Private Shared Sub DoDriveCheck()
+            ' Get Volume Label - valid drive
+            Dim serialNumber, maxFileNameLen, fileSystemFlags As Integer
+            Dim volNameLen As Integer = 50
+            Dim fileNameLen As Integer = 50
+            Dim volumeName As StringBuilder = New StringBuilder(volNameLen)
+            Dim fileSystemName As StringBuilder = New StringBuilder(fileNameLen)
+
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().First(Function(d) d.DriveType = IO.DriveType.Fixed)
+            Dim volumeInformationSuccess As Boolean = GetVolumeInformation(validDrive.Name, volumeName, volNameLen, serialNumber, maxFileNameLen, fileSystemFlags, fileSystemName, fileNameLen)
+
+            If volumeInformationSuccess Then
+                Assert.Equal(volumeName.ToString(), validDrive.VolumeLabel)
+            Else
+                Dim name As String = validDrive.VolumeLabel
+            End If
+        End Sub
+
+        <Fact>
+        <PlatformSpecific(TestPlatforms.Windows)>
```

Comment:
why this ? if these tests are just for windows ?

Reply:
The Platform Attributes come from the original .Net Tests. All the tests I wrote are platform independent. There are some tests that are platform specific today, DriveInfo is one of them. I did not write these but I assume all the test files run on all the platforms, the fact that it has .Windows. in the name is visual. I think these test also run on Linux but get skipped but I am not the expert here.